listener.go

```diff
@@ -101,7 +115,7 @@ func (l *Listener) ListenContext(ctx context.Context, network string, address st
 		service.Port, err = strconv.Atoi(port)
 
 		if err != nil {
-			err = fmt.Errorf("bad port number in network address %s://%s", network, address)
+			err = fmt.Errorf("bad port number in network address %s", service.Address)
```

Comment:
Was changing this intentional?

Reply:
Yeah, `network` and `address` where mistakenly used here since we're parsing `service.Address`. This is a bug that was highlighted because those variables were not available in the new method.